x/gasless/types/utils.go

```diff
@@ -0,0 +1,123 @@
+package types
+
+import (
+	"slices"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// Comparable is a type constraint that allows only comparable types.
+type Comparable interface {
+	comparable
+}
+
+// DeriveAddress derives an address with the given address length type, module name, and
+// address derivation name. It is used to derive private plan gas tank address.
+func DeriveAddress(addressType AddressType, moduleName, name string) sdk.AccAddress {
+	switch addressType {
+	case AddressType32Bytes:
+		return address.Module(moduleName, []byte(name))
+	case AddressType20Bytes:
+		return sdk.AccAddress(crypto.AddressHash([]byte(moduleName + name)))
+	default:
+		return sdk.AccAddress{}
+	}
+}
+
+func GetCoinByDenomFromCoins(denom string, coins sdk.Coins) (sdk.Coin, bool) {
+	for _, coin := range coins {
+		if coin.Denom == denom {
+			return coin, true
+		}
+	}
+	return sdk.Coin{}, false
+}
+
+// ItemExists returns true if item exists in array else false.
+func ItemExists[T Comparable](array []T, item T) bool {
+	return slices.Contains(array, item)
+}
+
+// RemoveDuplicates removes duplicates from a slice of any comparable type.
+func RemoveDuplicates[T Comparable](input []T) []T {
+	uniqueMap := make(map[T]bool)
+	var uniqueList []T
+	for _, v := range input {
+		if !uniqueMap[v] {
+			uniqueMap[v] = true
+			uniqueList = append(uniqueList, v)
+		}
+	}
+	return uniqueList
+}
+
+// RemoveValueFromList removes all occurrences of a specific value from a slice of any comparable type.
+func RemoveValueFromList[T Comparable](list []T, x T) []T {
+	return slices.DeleteFunc(list, func(v T) bool { return v == x })
+}
+
+func ShiftToEndUint64(list []uint64, x uint64) []uint64 {
```

Comment:
docstring needed and extract to own package

Reply:
extracted to slice_utils package